### PR TITLE
feat(serve): add autocompletion spec

### PR DESCRIPTION
**Pull request by Pkingo**
**What kind of change does this PR introduce? (Bug fix, feature, docs update, ...)**
feature: added autocompletion for serve

**What is the current behavior? (You can also link to an open issue here)**
None.

**What is the new behavior (if this is a feature change)?**
Added autocompletion for serve in accordance with their help section

**Additional info:**
This is my first contribution to fig, and I'm by no means an expert in serve. So if you have any suggestions or something you would like to see different, please let me know.

### CODE REVIEW

File: dev/serve.ts [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+
+      description:
+        "Specify a URI endpoint on which to listen - more than one may be specified to listen in multiple places",
+      args: {
+        name: "listen_uri",
+        suggestions: [
+          {
+            name: "port",
+            description: "E.g. serve --listen 1234",
+            type: "arg",
```
Comment (QuiiBz) [inline dev/serve.ts:28]: You can safely remove `type: "arg"` from all the suggestions

**Review by QuiiBz — CHANGES_REQUESTED** [review]
Thank you so much for this PR!

File: dev/serve.ts (hunk 1 above)
Comment (Pkingo) [reply dev/serve.ts]: Great. Thanks 💪

**Review by QuiiBz — APPROVED** [review]
LGTM, thanks a lot for this spec!

**Pull request merged**